src/components/View/View.tsx

```diff
@@ -71,3 +72,18 @@ function mapStateToProps(store: Store) {
 const ConnectedView = connect(mapStateToProps)(View)
 
 export default ConnectedView
+
+/**
+ * Add new values to `PopupWidgetTypes` from client application
+ *
+ * @param customWidgets
+ */
+function usePopupWidgetTypesExtension(customWidgets: Record<string, CustomWidgetDescriptor>) {
+    if (customWidgets) {
+        Object.entries(customWidgets).forEach(([widgetType, descriptor]) => {
+            if ((descriptor as CustomWidgetConfiguration).isPopup) {
```

Comment:
Use `isCustomWidget` typeguard:
https://github.com/tesler-platform/tesler-ui/blob/a5fdaca5e77958f9730b4a65aef4cbce4a820b9d/src/interfaces/widget.ts#L207

Reply:
> Use `isCustomWidget` typeguard:
> https://github.com/tesler-platform/tesler-ui/blob/a5fdaca5e77958f9730b4a65aef4cbce4a820b9d/src/interfaces/widget.ts#L207

So in this case condition will be look like this:
```typescript
if (!isCustomWidget(descriptor) && descriptor.isPopup)
```
And you'll call it "Divine condition", aren't you? =)
Like here:
https://github.com/tesler-platform/tesler-ui/blob/master/src/components/Widget/Widget.tsx#L64

Reply:
Personally I'd vote for `Godlike` but in referenced link the confusion was about variable named `customWidget` when in fact it's a descriptor which may or may not be custom widget.

Here it's named `descriptor` so no such confusion applied.